ansible/deploy.yml

```diff
@@ -185,11 +175,10 @@
           - "{{ local_peer_keys_dir }}/peer_key"
           - "{{ local_peer_keys_dir }}/peer_key.pub"
         when: stat_peer_dir.stat.exists
+        notify: "restart epoch daemon"
         tags: [config, peer_keys]
 
-      - name: Start epoch daemon
-        command: "{{ project_root }}/bin/epoch start"
-        tags: [daemon]
+      - meta: flush_handlers
```

Comment:
Do all handler blocks for `"start epoch daemon"` run then all those for `"restart epoch daemon"`? As if aggregated it would mean that ping is checked to return pong before starting node (that would fail).

Reply:
It's kind of pub/sub system and each handler is execute only once.

In regards to ping handler, [it never fails](https://github.com/aeternity/infrastructure/pull/94/files#diff-0a05600d18acbea9b59fd306c7e25899R249) it's used to pull the current state of the node.

Reply:
I've tested multiple scenarios and it works as expected, although I could be missing some edge case.

Reply:
Also for your connivance I ran the deploy on "empty" instances (no previous epoch deployed):
```
TASK [Fail on missing package variable] **************************************************************************************************************************************
Monday 10 September 2018  12:43:29 +0000 (0:00:00.765)       0:00:06.844 ****** 
skipping: [ae-dev1-epoch-n0]
skipping: [ae-dev1-epoch-n1]

TASK [Download remote package] ***********************************************************************************************************************************************
Monday 10 September 2018  12:43:29 +0000 (0:00:00.493)       0:00:07.337 ****** 
skipping: [ae-dev1-epoch-n0]
skipping: [ae-dev1-epoch-n1]

TASK [Upload local package] **************************************************************************************************************************************************
Monday 10 September 2018  12:43:30 +0000 (0:00:00.369)       0:00:07.706 ****** 
changed: [ae-dev1-epoch-n1]
changed: [ae-dev1-epoch-n0]

TASK [Make sure /home/epoch/node exists] *************************************************************************************************************************************
Monday 10 September 2018  12:44:52 +0000 (0:01:21.757)       0:01:29.464 ****** 
changed: [ae-dev1-epoch-n0]
changed: [ae-dev1-epoch-n1]

RUNNING HANDLER [Check epoch binary] *****************************************************************************************************************************************
Monday 10 September 2018  12:44:54 +0000 (0:00:02.017)       0:01:31.481 ****** 
ok: [ae-dev1-epoch-n1]
ok: [ae-dev1-epoch-n0]

RUNNING HANDLER [Ping epoch] *************************************************************************************************************************************************
Monday 10 September 2018  12:44:54 +0000 (0:00:00.739)       0:01:32.221 ****** 
skipping: [ae-dev1-epoch-n1]
skipping: [ae-dev1-epoch-n0]

RUNNING HANDLER [Stop epoch] *************************************************************************************************************************************************
Monday 10 September 2018  12:44:55 +0000 (0:00:00.467)       0:01:32.688 ****** 
skipping: [ae-dev1-epoch-n1]
skipping: [ae-dev1-epoch-n0]

TASK [Extract epoch package into /home/epoch/node] ***************************************************************************************************************************
Monday 10 September 2018  12:44:56 +0000 (0:00:01.111)       0:01:33.799 ****** 
changed: [ae-dev1-epoch-n0]
changed: [ae-dev1-epoch-n1]

TASK [Plant ENV file] ********************************************************************************************************************************************************
Monday 10 September 2018  12:45:03 +0000 (0:00:07.298)       0:01:41.098 ****** 
changed: [ae-dev1-epoch-n0]
changed: [ae-dev1-epoch-n1]

TASK [Make sure chain path ( ./db86.2 ) exists] ******************************************************************************************************************************
Monday 10 September 2018  12:45:06 +0000 (0:00:02.884)       0:01:43.983 ****** 
changed: [ae-dev1-epoch-n0]
changed: [ae-dev1-epoch-n1]

TASK [Make sure keys dir ( keys ) exists] ************************************************************************************************************************************
Monday 10 September 2018  12:45:08 +0000 (0:00:01.999)       0:01:45.983 ****** 
changed: [ae-dev1-epoch-n0]
changed: [ae-dev1-epoch-n1]

TASK [Copy node keys to /home/epoch/node/keys] *******************************************************************************************************************************
Monday 10 September 2018  12:45:10 +0000 (0:00:02.006)       0:01:47.990 ****** 
skipping: [ae-dev1-epoch-n0]
skipping: [ae-dev1-epoch-n1]

TASK [Update epoch configuration] ********************************************************************************************************************************************
Monday 10 September 2018  12:45:10 +0000 (0:00:00.383)       0:01:48.373 ****** 
changed: [ae-dev1-epoch-n0]
changed: [ae-dev1-epoch-n1]

TASK [Update accounts configuration] *****************************************************************************************************************************************
Monday 10 September 2018  12:45:14 +0000 (0:00:03.750)       0:01:52.124 ****** 
changed: [ae-dev1-epoch-n0]
changed: [ae-dev1-epoch-n1]

TASK [Check if peer keys directory exists] ***********************************************************************************************************************************
Monday 10 September 2018  12:45:17 +0000 (0:00:02.910)       0:01:55.035 ****** 
ok: [ae-dev1-epoch-n0]
ok: [ae-dev1-epoch-n1]

TASK [Copy node peer keys to /home/epoch/node/keys] **************************************************************************************************************************
Monday 10 September 2018  12:45:18 +0000 (0:00:00.750)       0:01:55.785 ****** 
skipping: [ae-dev1-epoch-n0] => (item=files/peer_keys/31.13.249.76/peer_key) 
skipping: [ae-dev1-epoch-n0] => (item=files/peer_keys/31.13.249.76/peer_key.pub) 
skipping: [ae-dev1-epoch-n1] => (item=files/peer_keys/31.13.249.75/peer_key) 
skipping: [ae-dev1-epoch-n1] => (item=files/peer_keys/31.13.249.75/peer_key.pub) 

RUNNING HANDLER [Check epoch binary] *****************************************************************************************************************************************
Monday 10 September 2018  12:45:18 +0000 (0:00:00.223)       0:01:56.009 ****** 
ok: [ae-dev1-epoch-n0]
ok: [ae-dev1-epoch-n1]

RUNNING HANDLER [Ping epoch] *************************************************************************************************************************************************
Monday 10 September 2018  12:45:19 +0000 (0:00:00.947)       0:01:56.956 ****** 
changed: [ae-dev1-epoch-n0]
changed: [ae-dev1-epoch-n1]

RUNNING HANDLER [Stop epoch] *************************************************************************************************************************************************
Monday 10 September 2018  12:45:22 +0000 (0:00:02.591)       0:01:59.548 ****** 
skipping: [ae-dev1-epoch-n0]
skipping: [ae-dev1-epoch-n1]

RUNNING HANDLER [Start epoch] ************************************************************************************************************************************************
Monday 10 September 2018  12:45:22 +0000 (0:00:00.359)       0:01:59.907 ****** 
changed: [ae-dev1-epoch-n0]
changed: [ae-dev1-epoch-n1]
```